### ISSUE TITLE
Can user be removed from online meeting after transfer or can the online meeting be closed by IncidentBot?

### ISSUE THREAD

**Issue by rajlpand** [microsoftgraph/microsoft-graph-comms-samples#92]
Hi,
 
I had a question regarding the IncidentBot sample under:
 
https://github.com/microsoftgraph/microsoft-graph-comms-samples/tree/master/Samples/BetaSamples/RemoteMediaSamples
 
This sample works great and it is creating incidents followed by adding user to online meeting. I wanted to check if it is possible to remove the user from the online meeting they were added to earlier via IncidentBot? 
 
I tried ending the call using ICall.DeleteAsync created at:
 
https://github.com/microsoftgraph/microsoft-graph-comms-samples/blob/9db883a1e8649be9889f8ed500969634695b3bcd/Samples/BetaSamples/RemoteMediaSamples/IncidentBot/Bot/Bot.cs#L214
 
However looks like that call got deleted when the user was transferred to incident meeting, appreciate any thoughts or comments on this. Additional question about this, can the IncidentBot close/end the online meeting it has joined/started?

**Comment by ksikorsk** [comment]
Can you remind me of the flow please?  Sample calls a couple agents, then when they pick up they get transferred to the meeting?

**Comment by rajlpand** [comment]
Sure the flow I am using is to raise an incident, here are the steps in this flow:
1. user sends a HTTP POST request to raise incident
2. to process that request IncidentBot joins online meeting
3. IncidentBot makes a call to user
4. when user answers call it plays prompt to press 1 or 0
5. if user presses 1 IncidentBot adds user to online meeting

**Comment by ksikorsk** [comment]
If you can send me some logs, that would help as well.  Best thing would be a single http headers snapshot of an incoming notification.

**Comment by ksikorsk** [comment]
For 5) I don't see any usage of `TransferAsync`, only `InviteAsync`.  Invite will send a new invite to the user, but will not terminate the original call.  If the original call to the user is a P2P call, the call will end if any of the participants leave.

**Comment by ksikorsk** [comment]
Apologies, the sample is doing things correctly... `InviteAsync` specifies `replacesCallId`, which will move the user from the original call into the meeting.

**Comment by ksikorsk** [comment]
At which point the original meeting is terminated for all parties, you need not delete it.  You should have a notification raised that the original call is `terminated` and then removed from `ICallCollection`

**Comment by rajlpand** [comment]
Yes, it looks like original call is deleted when user is added to meeting at:
https://github.com/microsoftgraph/microsoft-graph-comms-samples/blob/9db883a1e8649be9889f8ed500969634695b3bcd/Samples/BetaSamples/RemoteMediaSamples/IncidentBot/Bot/MeetingCallHandler.cs#L63

I am trying out a case where once the user is done with the online meeting if IncidentBot can end the online meeting it started/joined or if it can remove the user from online meeting?

**Comment by ksikorsk** [comment]
A couple terminology clarifications:
when you say online meeting, you mean the multiparty call?  We have 2 different concepts: online meeting, and call.  The online meeting is usually just the coordinates on how people can connect to each other, from there a call is created with the online meeting coordinates.  Same online meeting can have many multiparty calls.  Each call is ended once all participants leave, and started again once a participant joins.

We do not have any APIs to eject participants from multiparty calls.  Today we only support ending P2P calls.  What we do have a flag `AllowConversationWithoutHost` where **if the bot is the first one** to join the meeting, and it leaves, **everyone is removed from the meeting**.

**Comment by rajlpand** [comment]
Thanks for the clarification and explanation, if I understood it then as of now there is no API to end multiparty call. I would like to request that as an enhancement where a bot can end multiparty call via API that it started, please let me know if I need to create some ticket/request for that or if this issue can be used for that purpose.

**Comment by ksikorsk** [comment]
If the bot always joins first, you can use `AllowConversationWithoutHost` as mentioned below.  But if you want to join a meeting then end it, you cannot.

@VinodRavichandran Request for an API to terminate a call for every participant.
